Blueprint - List non active plugins and themes

moon0326 opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes #56082 .

Previously, Blueprint listed only active plugins and themes. This PR now includes inactive ones as well, but they are deselected by default.

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
|   ![Screenshot 2025-03-05 at 10 54 33 AM](https://github.com/user-attachments/assets/93f49642-c345-43df-b4f5-56a82c753c9c)     |   ![Screenshot 2025-03-05 at 10 54 55 AM](https://github.com/user-attachments/assets/dd5155a1-2c64-45f6-8941-b0a3909b149c)    |


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch. 
2. Navigate to `WooCommerce -> Settings -> Advanced -> Features and enable Blueprint (beta)`
3. Navigate to `Plugins` and ensure at least one plugin is inactive.
4. Navigate to `WooCommerce -> Settings -> Advanced -> Bluerpint (beta)`
5. Click on `Plugins and extensions`
6. Confirm the inactive plugin is listed and de-selected.
7. Navigate to `Themes` and install an extra theme.
8. Navigate to `WooCommerce -> Settings -> Advanced -> Bluerpint (beta)`
9. Click on "Themes"
10. Confirm inactive themes are de-selected.
<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Blueprint: Lists non-active themes and plugins.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

Commit:
blueprint js settings should be added only on the blueprint page

Commit:
List non-active plugins and themes

Commit:
Ensure settings are checked by default

ilyasfoo commented on plugins/woocommerce/src/Admin/Features/Blueprint/Init.php:
```diff
@@ -116,10 +116,41 @@ public function add_woo_exporters( array $exporters ) {
+			},
+			$themes
+		);
+
```
For consistency, what do you think of sorting `$themes` here similar to `$plugins`

ilyasfoo reviewed (approved):
Testing well! I have a few feedback, but pre-approving:

- Changelog required
- Not a blocker for this PR, but I think it'll be good to start adding unit tests for `get_step_groups_for_js`. I'd reckon an AI would do decent job generating a basic coverage for starters!

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | Commented |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | ✅ All good |
| Changelog  | Commented |
| Documentation  | Not applicable |
| PR description  | ✅ All good |
| Testing instructions  | ✅ All good |
| Manual testing  | ✅ Works as expected |

#### Scenarios tested

| Scenario  | Result | Screenshot |
| ------------- | ------------- | ------------- |
| List non-active plugins and themes  | ✅ Functionality worked correctly | <img width="909" alt="image" src="https://github.com/user-attachments/assets/d73ecfa4-1dfc-487e-9c86-a0cb57e12109" /> |
| Import blueprint JSON  | ✅ Functionality worked correctly | |
| Export blueprint JSON  | ✅ Functionality worked correctly | |

Commit:
Refactor get_step_groups_for_js and add unit tests

moon0326 commented:
> Testing well! I have a few feedback, but pre-approving:

Thank you 🙇 

I've refactored the function a bit and added unit tests [Refactor get_step_groups_for_js and add unit tests](https://github.com/woocommerce/woocommerce/pull/56141/commits/b0dd7cf1809894f152828fd47f2bc180032deec7)

This commit also sorts $themes as suggested.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce, woocommerce/client/admin

adrianduffell reviewed (approved):
Tested well on JN per instructions. Thanks @moon0326 🚀 

<img width="1792" alt="inactive theme plugin" src="https://github.com/user-attachments/assets/8a207638-139f-4078-843a-a1a8406cd811" />

Commit:
Merge branch 'trunk' into update/56082-list-non-active-plugins-and-themes-de-selected
